lib/utils/fetch.ts

```diff
@@ -1,38 +1,59 @@
 import { Auth } from 'aws-amplify';
 
-// https://eckertalex.dev/blog/typescript-fetch-wrapper
-async function http<T>(path: string, config?: RequestInit): Promise<T> {
+export type ResponseWithStatus<T> = { data: T; isSuccess: true } | { isSuccess: false; errorMessage: string };
```

Comment:
I've redesigned this. I don't think it's a good idea to hide the asynchronous nature of requests into SWR since we lose control over the way promises are handled.